examples/consumer-listener/consumer-listener.go

```diff
@@ -34,15 +34,9 @@ func main() {
 
 	channel := make(chan pulsar.ConsumerMessage, 100)
 
-	options := pulsar.ConsumerOptions{
-		Topic:            "topic-1",
-		SubscriptionName: "my-subscription",
-		Type:             pulsar.Shared,
-	}
-
-	options.MessageChannel = channel
-
-	consumer, err := client.Subscribe(options)
+	consumer, err := client.Subscribe("my-subscription", pulsar.WithTopics("topic-1"),
+		pulsar.WithSubscriptionType(pulsar.Shared),
+		pulsar.WithMessageChannel(channel))
```

Comment:
To me it seems the proposed syntax is more verbose and less descriptive at the same time. For example, given a different name, it might become more difficult that `"my-subscription"` is a subscription name and not a topic name.

Reply:
@merlimat  I know ur question, we could use WithSybcriptionName to expose the set of subcription name.